src/Admin/MultisitesModelAdminExtension.php

```diff
@@ -153,27 +171,47 @@ public function getActiveSiteSessionKey() {
 		return 'Multisites_ActiveSite_For_' . get_class($this->owner);
 	}
 
-
-	/**
-	 * Get and cache an instance of the data class currently being listed
-	 *
-	 * @return DataObject
-	 **/
-	private function getListDataClass() {
-		if(!$this->listDataClass) {
-			$this->listDataClass = singleton($this->owner->getSearchContext()->getResults(array())->dataClass());
-		}		
-		return $this->listDataClass;
-	}
-
-
-	/**
-	 * Determines whether the current model being managed is MultiSitesAware
-	 *
-	 * @return boolean
-	 **/
-	private function modelIsMultiSitesAware() {
-		$model = $this->getListDataClass();
-		return $model->hasExtension('MultisitesAware') || $model instanceof SiteTree;
-	}
+    /**
+     * Get and cache an instance of the data class currently being listed
+     *
+     * @return DataObject
+     **/
+    private function getListDataClass() {
+        /**
+         * Silverstripe-recipe 4.3.0 introduces changes that broke the
+         * previous approach to retrieving the model class.
+         *
+         * This is/will be fixed in silverstripe-recipe 4.3.1
+         *
+         * We've taken this approach so that Multisites can work with
+         * SS 4.3.0 so long as you don't use the MultisitesAware extension.
+         */
+        if ($this->listDataClass === null) {
+            // only true if silverstripe-recipe > 4.3.0
+            if ($this->owner->hasMethod('getModelClass')) {
+                $this->listDataClass = $this->owner->getModelClass();
+            }
+            // else false to bypass the MultisitesAware extension
+            else {
+                $this->listDataClass = false;
+            }
+        }
+        return $this->listDataClass;
+    }
+
+    /**
+     * Determines whether the current model being managed is MultiSitesAware
+     *
+     * @return boolean
+     **/
+    private function modelIsMultiSitesAware() {
+        $model = $this->getListDataClass();
+        // always false if using silverstripe-recipe <= 4.3.0
+        if ($model) {
+            $isAware = $model::has_extension(MultisitesAware::class);
+            $isSiteTree = in_array(SiteTree::class, ClassInfo::ancestry($model));
```

Comment:
@JoshuaCarter does https://api.silverstripe.org/3/ClassInfo.html#method_subclassesFor exist in SS4? That's the one I was thinking of, but it's possible they removed it?

Reply:
Discussed this one, ancestry is the one we want :)